v2/contribute/decisions/sessionclaims/0003-claim-redirection-targets-on-fe.mdx

```diff
@@ -0,0 +1,34 @@
+---
+id: "0003"
+title: Claim redirection functions should return full urls
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Claim redirection functions should return full urls
+
+<DecisionHeader status="proposed" lastUpdate="2022-11-28" created="2022-11-28" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+We have decided to use configurable callbacks to redirect on claim validation success/failure (see [here](./0001)). These functions determine where we redirect the user by returning a string. We have to decide if this string is a full URL or just a path
+
+## Considered Options
+
+* Path
+* Full URL
+
+## Decision Outcome
+
+The return value should be taken interpreted as a full URL (with the option to return just the path for local redirections):
+
+- This doesn't matter if there is no session sharing between subdomains. This is the case for most sites.
+- It makes sense for `onFailure` to redirect to the website domain in many cases (e.g: email verification check failing)
+- There are cases, where `onFailure` doesn't point to the website domain, e.g.: the access denied page is specific to the app/subdomain we are currently on
+- `onSuccess` is even more likely to redirect to a domain that doesn't match the websitedomain (e.g: continue browsing after email verification/2FA on websitedomain)
+- Even if we could make onFailure redirect only to the website domain, doing the same to `onSuccess` would block too many usecases (or cause awkward double redirections).
```

Comment:
Should we also allow onRedirectionUrl callbacks to return full domains in this case? Cause right now, they are only allowed to return a path.

Reply:
`getRedirectUrl` can return full URLs, the `with-one-login-many-subdomains` example uses this.